backend/chromeApiServer/swagger.yaml

```diff
@@ -0,0 +1,222 @@
+          in: query
+          description: ID of student to search for submission
+          required: true
+          schema:
+            type: string
+        - name: userType
+          in: query
+          description: Type of user for the action. This is an ENUM, so enter either STUDENT or GRADER
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: object
+
+        '400':
+          description: Invalid parameter entered
+        '401':
+          description: UserType not Student, unauthorized access
+
+  /evaluate:
+    post:
+      tags:
+        - student
+      summary: evaluate the student submission
+      description: Starts the evaluation process to ensure the student assignment compiles with the makefile
+      operationId: evaluateCode
+      parameters:
+        - name: Authorization
+          in: header
+          description: bearer token for Oauth validation
+          required: true
+          schema:
+            type: string
+        - name: courseId
+          in: query
+          description: ID of course to search for submission.Pulled from url
+          required: true
+          schema:
+            type: string
+        - name: assignmentId
+          in: query
+          description: ID of assignment in a course to search for submission. Pulled from url
+          required: true
+          schema:
+            type: string
+        - name: userType
+          in: query
+          description: Type of user for the action. This is an ENUM, so enter either STUDENT or GRADER
+          required: true
+          schema:
+            type: string
+      #TODO: Update this section with example
+      requestBody:
+        content:
+          multipart/form-data: # Media type
+            schema:            # Request payload
+              type: object
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: object
+        '400':
+          description: Invalid parameter entered
+        '401':
+          description: UserType not Grader, unauthorize access
+
+  /oauth2Response:
+    get:
+      tags:
+        - oauth
+      summary: Request to fetch auth token
+      description: Initiates the oauth login process and fetches the access token from Canvas
+      operationId: oauth
+      parameters:
+        - name: code
+          in: query
+          description: bearer token for Oauth validation
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: successful operation
+        '401':
+          description: UserType not Grader, unauthorize access
+#TODO: update this section with examples of our responses
+components:
```

Comment:
are you planning on updating the components section?

Reply:
Yes I do, this is the first draft. I needed to get the data first.

Reply:
Updated the swagger doc with the correct components